Gem/Code/Source/Components/UI/UiMatchPlayerCoinCountsComponent.cpp

```diff
@@ -6,87 +6,86 @@
-#endif
     }
 
-#if AZ_TRAIT_CLIENT
-    void UiMatchPlayerCoinCountsComponent::OnPressed([[maybe_unused]] float value)
+    void UiMatchPlayerCoinCountsComponent::UpdatePlayerScoreUI()
     {
-        const StartingPointInput::InputEventNotificationId* inputId = StartingPointInput::InputEventNotificationBus::GetCurrentBusId();
-
-        if (inputId == nullptr)
+        AZStd::vector<PlayerCoinState> coins = AZ::Interface<MatchPlayerCoinsComponent>::Get()->GetPlayerCoinCounts();
+        AZStd::size_t elementIndex = 0;
+        for (const PlayerCoinState& state : coins)
         {
-            return;
-        }
-
-        if (*inputId == ShowPlayerCoinCountsEventId)
-        {
-            UiElementBus::Event(m_rootElementId, &UiElementBus::Events::SetIsEnabled, true);
-
-            AZStd::vector<PlayerCoinState> coins;
-            MatchPlayerCoinsRequestBus::BroadcastResult(coins, &MatchPlayerCoinsRequestBus::Events::GetPlayerCoinCounts);
-            AZStd::size_t elementIndex = 0;
-            for (const PlayerCoinState& state : coins)
+            if (elementIndex >= m_playerRowElement.size())
             {
-                if (elementIndex >= m_playerRowElement.size())
-                {
-                    break;
-                }
-
-                if (state.m_playerId != Multiplayer::InvalidNetEntityId)
-                {
-                    AZStd::vector<AZ::EntityId> children;
-                    UiElementBus::EventResult(children, m_playerRowElement[elementIndex], &UiElementBus::Events::GetChildEntityIds);
-                    if (children.size() >= 2)
-                    {
-                        const PlayerNameString name = GetPlayerName(state.m_playerId);
-                        UiTextBus::Event(children[0], &UiTextBus::Events::SetText, AZStd::string::format("%s", name.c_str()));
-                        UiTextBus::Event(children[1], &UiTextBus::Events::SetText, AZStd::string::format("%d", state.m_coins));
-                    }
-                    elementIndex++;
-                }
+                break;
             }
 
-            // Clear out the unused fields.
-            for (; elementIndex < m_playerRowElement.size(); ++elementIndex)
+            if (state.m_playerId != Multiplayer::InvalidNetEntityId)
             {
                 AZStd::vector<AZ::EntityId> children;
                 UiElementBus::EventResult(children, m_playerRowElement[elementIndex], &UiElementBus::Events::GetChildEntityIds);
                 if (children.size() >= 2)
                 {
-                    UiTextBus::Event(children[0], &UiTextBus::Events::SetText, "");
-                    UiTextBus::Event(children[1], &UiTextBus::Events::SetText, "");
+                    const PlayerNameString name = GetPlayerName(state.m_playerId);
+                    UiTextBus::Event(children[0], &UiTextBus::Events::SetText, AZStd::string::format("%s", name.c_str()));
```

Comment:
Suggestion: You can probably simplify `AZStd::string::format("%s", name.c_str())` to either `AZStd::string(name)` or AZStd::string(name.c_str())`

Reply:
Ah, fixed! Thank you!